internal/server/handler/metric.go

```diff
@@ -23,16 +27,24 @@ func (h *MetricHandlers) Update(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	mName := chi.URLParam(r, "name")
-	mValue, err := strconv.ParseFloat(chi.URLParam(r, "value"), 64)
+	mName := r.PathValue("name")
+	mValue, err := strconv.ParseFloat(r.PathValue("value"), 64)
 
 	if err != nil {
 		log.Printf("handler Update metrics fail: %s", err.Error())
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	if err := h.Storage.Save(mType, mName, mValue); err != nil {
+	ival := int64(mValue)
```

Comment:
В Go не принято обозначать тип перменной через суффикс.

```suggestion
	delta := int64(mValue)
```